platform.js

```diff
@@ -403,5 +416,23 @@ module.exports = {
     }
 
     return module.exports.spawn(command, args);
+  },
+  listDirectory(basePath) {
+    return new Promise((resolve, reject)=>{
+      var response = [];
+
+      fs.readdir(basePath, (err, items)=>{
+        if (err) { reject(err); }
+
+        items.forEach((item)=>{
+          var itemPath = path.join(basePath, item);
+          var stat = fs.statSync(itemPath);
```

Comment:
@fjvallarino I'm not really comfortable mixing sync and async calls in the same promise handler.